templates/frontend/objects/article_summary.tpl

```diff
@@ -60,6 +61,12 @@
 	{if $hasAccess}
 		<ul class="galleys_links">
 			{foreach from=$article->getGalleys() item=galley}
+				{if $primaryGenreIds}
+					{assign var="file" value=$galley->getFile()}
+					{if !$file || !in_array($file->getGenreId(), $primaryGenreIds)}
+						{php}continue;{/php}
+					{/if}
+				{/if}
```

Comment:
The `{$primaryGenreIds}` variable isn't retrieved from the submission files. It's taken directly from the `GenreDAO`, so as long as a journal has any file genre that counts as "primary", `{$primaryGenreIds}` will be populated.

I'm not sure what a journal without a primary file genre would be doing. But I wouldn't want to assume that they want to display supplementary files on the table of contents, so I'd be tempted to do nothing for that edge case.

Did that answer your question? Or were you asking about just a single article with no primary files?

In that case, the each loop should `continue` every time, and display no files in the article summary. We'll end up with an empty `<ul>`, but that doesn't seem like a big problem for what will likely be a rare edge case.